### PR TITLE
emqx self hosted runners

### PR DESCRIPTION
**Pull request by id**
- [x] use redis to store runner config instead of ssm to avoid throttling
- [ ] arm64 runners
- [ ] docker proxy
- [ ] use ssm for ami id

Reference PR using new setup in my dummy org: https://github.com/id-org/emqx/pull/1

### CODE REVIEW

File: emqx/main.tf [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  enable_ssm_on_runners = true
+  enable_ephemeral_runners = true
+  enable_organization_runners = true
+  enable_job_queued_check = true
+  enable_fifo_build_queue = true
+  runner_run_as = "ubuntu"
+  delay_webhook_event = 0
+  minimum_running_time_in_minutes = 5
+  runners_maximum_count = 256
+  scale_down_schedule_expression = "cron(* * * * ? *)"
```
Comment (thalesmg) [inline emqx/main.tf]: Q: is this a [valid cron expression](https://crontab.guru/#*_*_*_*_?_*)?

File: emqx/providers.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+provider "aws" {
+  region = local.aws_region
+
+  default_tags {
+    tags = {
+      Example = local.environment
```
Comment (thalesmg) [inline emqx/providers.tf]: Q: should the tag name indeed be `Example`?  Perhaps `Env = local.environment`?

File: images/emqx-amd64/base.sh [hunk 3]
```diff
@@ -0,0 +1,124 @@
+#!/bin/bash -e
+
+export DEBIAN_FRONTEND=noninteractive
+
+# Enable retry logic for apt up to 10 times
+echo "APT::Acquire::Retries \"10\";" > /etc/apt/apt.conf.d/80-retries
+
+# Configure apt to always assume Y
+echo "APT::Get::Assume-Yes \"true\";" > /etc/apt/apt.conf.d/90assumeyes
```
Comment (thalesmg) [inline images/emqx-amd64/base.sh]:
```suggestion
echo "APT::Get::Assume-Yes \"true\";" > /etc/apt/apt.conf.d/90-assumeyes
```
?

**Review by thalesmg — APPROVED** [review]

File: emqx/main.tf (hunk 1 above)
Comment (id) [reply emqx/main.tf]: It is for lambda, apparently https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents-expressions.html

**Review by qzhuyan — APPROVED** [review]

**Pull request merged**